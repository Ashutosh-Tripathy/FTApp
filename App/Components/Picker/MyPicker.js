import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View, Text, Picker } from 'react-native'

function MyPicker (props) {
  let items = [{label: 'Please select any value', value: 0}, ...props.items];
  return (
      <View>
      <Picker
      selectedValue={props.selectedValue}
      style={{ height: 50, width: 100 }}
      onValueChange={(itemValue, itemIndex) => props.onValueChange(itemValue, itemIndex)}>
      {
        items.map(item => <Picker.Item key={item.value} label={item.label} value={item.value} />)
      }
      </Picker>
      </View>)
}

MyPicker.propTypes = {
  selectedValue: PropTypes.number.isRequired,
  items: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.number.isRequired
  })).isRequired
}

export default MyPicker
