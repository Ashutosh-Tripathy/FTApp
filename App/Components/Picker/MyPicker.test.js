
import React from 'react';
import { shallow, mount } from 'enzyme';
import MyPicker from './MyPicker';
import jsdom from 'jsdom';

describe('MyComponent', () => {
  it('should render correctly', () => {
    const component = shallow(<MyPicker selectedValue='0' />);

    expect(component).toMatchSnapshot();
  });


  it('should call click event', () => {
    //const onPress = jest.fn();
    //debugger;
    //const component = shallow(<Picker title='Default title' onPress={onPress} />);
    //let btn = component;
    //console.log(btn);
    //let a = 'TTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT';
    //console.log(a);
    //component.find(Picker).first().props().onPress();
    //expect(onPress).toHaveBeenCalled();
  });
});
