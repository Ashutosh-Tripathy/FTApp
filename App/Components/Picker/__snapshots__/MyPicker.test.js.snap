// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyPicker
    items={
      Array [
        Object {
          "label": "label1",
          "value": 1,
        },
        Object {
          "label": "label2",
          "value": 2,
        },
      ]
    }
    onValueChange={[MockFunction]}
    selectedValue={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Picker
        mode="dialog"
        onValueChange={[Function]}
        selectedValue={0}
        style={
          Object {
            "height": 50,
            "width": 100,
          }
        }
      >
        <PickerItem
          label="label1"
          value={1}
        />
        <PickerItem
          label="label2"
          value={2}
        />
      </Picker>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <PickerItem
            label="label1"
            value={1}
          />,
          <PickerItem
            label="label2"
            value={2}
          />,
        ],
        "mode": "dialog",
        "onValueChange": [Function],
        "selectedValue": 0,
        "style": Object {
          "height": 50,
          "width": 100,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "label": "label1",
            "value": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "label": "label2",
            "value": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Picker
          mode="dialog"
          onValueChange={[Function]}
          selectedValue={0}
          style={
            Object {
              "height": 50,
              "width": 100,
            }
          }
        >
          <PickerItem
            label="label1"
            value={1}
          />
          <PickerItem
            label="label2"
            value={2}
          />
        </Picker>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <PickerItem
              label="label1"
              value={1}
            />,
            <PickerItem
              label="label2"
              value={2}
            />,
          ],
          "mode": "dialog",
          "onValueChange": [Function],
          "selectedValue": 0,
          "style": Object {
            "height": 50,
            "width": 100,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "label": "label1",
              "value": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "label": "label2",
              "value": 2,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
